### PR TITLE
New keyboard uranuma

### PR DESCRIPTION
**Pull request by yohewi**
## New keyboard UraNuma

 It is a new small keyboard.
 It is a keyboard to input using fingers and palms.
 There are 46 keys.
 

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR
* 
## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/uranuma/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,80 @@
+#include QMK_KEYBOARD_H
+
+enum layer_number {
+  _BASE,
+  _LOWER,
+  _RAISE,
+  _ADJUST,
```
Comment (noroadsleft) [inline keyboards/uranuma/keymaps/default/keymap.c]: You don't have an `_ADJUST` layer here.

```suggestion
```

File: keyboards/uranuma/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+};
+
+enum custom_keycodes {
+  BASE = SAFE_RANGE,
+  KANJI,
+  LOWER,
+  RAISE,
+};
```
Comment (noroadsleft) [inline keyboards/uranuma/keymaps/default/keymap.c]: `KANJI` is the only keycode you're using here, and it doesn't have any functionality written.

```suggestion
enum custom_keycodes {
  KANJI = SAFE_RANGE,
};
```

File: keyboards/uranuma/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+// Layer Mode aliases
+#define KC_LOWER LOWER
+#define KC_RAISE RAISE
+#define KC_ADUJST ADJUST
+#define KC_DLBS BASE //DF(_BASE)
+
+#define KC_RST   RESET
+#define KC_SUSFT SUSFT
```
Comment (noroadsleft) [inline keyboards/uranuma/keymaps/default/keymap.c]: You aren't using any of these.

```suggestion
```

File: keyboards/uranuma/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+#define KC_MNSF  LSFT_T(KC_MINS)
+#define KC_ESCT  LCTL_T(KC_ESC)
+#define KC_TBAL  LALT_T(KC_TAB)
+#define KC_ALAL  LALT_T(KC_A)
+#define KC_ALAP  LALT_T(KC_APP)
+#define KC_JEQL  LSFT(KC_MINS)
+#define KC_WLCK  LALT(KC_L)
+#define KC_SFEQ  LSFT(KC_MINS)
+#define KC_SFPL  LSFT(KC_SCLN)
+#define KC_SFAS  LSFT(KC_QUOT)
```
Comment (noroadsleft) [inline keyboards/uranuma/keymaps/default/keymap.c]: You aren't using any of these either.

```suggestion
```

File: keyboards/uranuma/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+      KC_Q,  KC_W,  KC_E,  KC_R,  KC_T,                    KC_Y, KC_U, KC_I, KC_O, KC_P,
+  //|------+------+------+------+------|                |------+------+------+------+-----
+      KC_A,  KC_S, KC_D,  KC_F, KC_G,                KC_H,  KC_J,  KC_K, KC_L,  KC_BSLS,
+  //|------+------+------+------+------|                |------+------+------+------+-----
+      KC_ZSFT, KC_X, KC_C, KC_V, KC_B,               KC_N, KC_M, KC_COMM, KC_DOT, KC_MNSF,
+  //|------+------+------+------+------|                |------+------+------+------+------
+      KC_ESCT, KC_TBAL, KC_PSCR, KC_LALT, KC_SPC, KC_ENT, LALT(KC_GRV), KC_SLSH, KC_BSLS, KC_RBRC,
+  //|------+------+------+------+------|                |------+------+------+------+------
+      KC_LSFT, MO(_LOWER),               KC_BSPC, KC_ENT,          MO(_RAISE),KC_RCTL
+  //|------+------+------+------+------|                |------+------+------+------+------
```
Comment (noroadsleft) [inline keyboards/uranuma/keymaps/default/keymap.c]: Just a question here:

Would this be a reasonably accurate representation of this keymap for QMK Configurator?

image

Comment (yohewi) [reply keyboards/uranuma/keymaps/default/keymap.c]: Yes, you are right!

File: keyboards/uranuma/uranuma.h [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+  k30, k31, k32, k33, k34,     k35, k36, k37, k38, k39, \
+  k40, k41,           k42,     k43,           k44, k45  \
+  ) \
+{ \
+	{ k00, k01, k02, k03, k04,   k05, k06, k07, k08, k09 }, \
+	{ k10, k11, k12, k13, k14,   k15, k16, k17, k18, k19 }, \
+	{ k20, k21, k22, k23, k24,   k25, k26, k27, k28, k29 }, \
+	{ k30, k31, k32, k33, k34,   k35, k36, k37, k38, k39 }, \
+    { k40, k41, KC_NO, KC_NO, k42, k43, KC_NO, KC_NO, k44, k45 } \
+}
```
Comment (noroadsleft) [inline keyboards/uranuma/uranuma.h]:
```suggestion
#define LAYOUT( \
    k00, k01, k02, k03, k04,               k05, k06, k07, k08, k09, \
    k10, k11, k12, k13, k14,               k15, k16, k17, k18, k19, \
    k20, k21, k22, k23, k24,               k25, k26, k27, k28, k29, \
    k30, k31, k32, k33, k34, k44,     k45, k35, k36, k37, k38, k39, \
    k40, k41,                                             k48, k49  \
) { \
    { k00, k01, k02,   k03,   k04, k05, k06,   k07,   k08, k09 }, \
    { k10, k11, k12,   k13,   k14, k15, k16,   k17,   k18, k19 }, \
    { k20, k21, k22,   k23,   k24, k25, k26,   k27,   k28, k29 }, \
    { k30, k31, k32,   k33,   k34, k35, k36,   k37,   k38, k39 }, \
    { k40, k41, KC_NO, KC_NO, k44, k45, KC_NO, KC_NO, k48, k49 }  \
}
```

File: keyboards/uranuma/keymaps/default/keymap.c [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+enum custom_keycodes {
+  KANJI = SAFE_RANGE,
+};
+
+// Layer Mode aliases
+#define KC_ZSFT  LSFT_T(KC_Z)
+#define KC_MNSF  LSFT_T(KC_MINS)
+#define KC_ESCT  LCTL_T(KC_ESC)
+#define KC_TBAL  LALT_T(KC_TAB)
+
```
Comment (noroadsleft) [inline keyboards/uranuma/keymaps/default/keymap.c]:
```suggestion
#define ALT_GRV  LALT(KC_GRV)
#define LOWER    MO(_LOWER)
#define RAISE    MO(_RAISE)

```

File: keyboards/uranuma/keymaps/default/keymap.c [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+  //|---------+---------+---------+---------+---------+---------+---------+---------+---------+---------
+       KC_F1,    KC_F2,    KC_F3,    KC_F4,   KC_F5,  KC_F6,    KC_F7,    KC_F8,    KC_F9,   KC_F10,
+  //|---------+---------+---------+---------+---------+---------+---------+---------+---------+---------
+       KC_F11,  KC_F12, XXXXXXX,  KANJI,  XXXXXXX,  XXXXXXX,  XXXXXXX,  XXXXXXX,  KC_UP,    KC_RO,
+  //|---------+---------+---------+---------+---------+---------+---------+---------+---------+---------
+      KC_ESCT,KC_TBAL,KC_LGUI, MO(_LOWER),KC_BSPC, KC_ENT, KANJI, KC_LEFT, KC_DOWN, KC_RGHT,
+  //|------+------+------+------+------|                |------+------+------+------+------|
+      KC_RSFT, MO(_LOWER),               KC_SPC, KC_ENT,                MO(_RAISE), KC_RCTL
+  //|------+------+------+------+------|                |------+------+------+------+------|
+  ),
```
Comment (noroadsleft) [inline keyboards/uranuma/keymaps/default/keymap.c]:
```suggestion
    [_BASE] = LAYOUT(
     //.--------------------------------------------.                     .--------------------------------------------.
        KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,                          KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,
     //|--------+--------+--------+--------+--------|                     |--------+--------+--------+--------+--------|
        KC_A,    KC_S,    KC_D,    KC_F,    KC_G,                          KC_H,    KC_J,    KC_K,    KC_L,    KC_BSLS,
     //|--------+--------+--------+--------+--------|                     |--------+--------+--------+--------+--------|
        KC_ZSFT, KC_X,    KC_C,    KC_V,    KC_B,                          KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_MNSF,
     //|--------+--------+--------+--------+--------+--------.   .--------+--------+--------+--------+--------+--------|
        KC_ESCT, KC_TBAL, KC_PSCR, KC_LALT, KC_SPC,  KC_BSPC,     KC_ENT,  KC_ENT,  ALT_GRV, KC_SLSH, KC_BSLS, KC_RBRC,
     //|--------+--------+--------+--------+--------+--------'   '--------+--------+--------+--------+--------+--------|
        KC_LSFT, LOWER,                                                                               RAISE,   KC_RCTL
     //'-----------------'                                                                           '-----------------'
    ),

    [_LOWER] = LAYOUT(
     //.--------------------------------------------.                     .--------------------------------------------.
        KC_1,    KC_2,    KC_3,    KC_4,    KC_5,                          KC_6,    KC_7,    KC_8,    KC_9,    KC_0,
     //|--------+--------+--------+--------+--------|                     |--------+--------+--------+--------+--------|
        KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,                         KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,
     //|--------+--------+--------+--------+--------|                     |--------+--------+--------+--------+--------|
        KC_F11,  KC_F12,  XXXXXXX, XXXXXXX, XXXXXXX,                       XXXXXXX, XXXXXXX, KC_SCLN, KC_UP,   KC_EQL,
     //|--------+--------+--------+--------+--------+--------.   .--------+--------+--------+--------+--------+--------|
        KC_ESCT, KC_TBAL, KC_LGUI, KC_ENT,  KC_BSPC, KC_SPC,      KC_ENT,  KC_LALT, KANJI,   KC_LEFT, KC_DOWN, KC_RGHT,
     //|--------+--------+--------+--------+--------+--------'   '--------+--------+--------+--------+--------+--------|
        KC_LSFT, LOWER,                                                                               RAISE,   KC_RCTL
     //'-----------------'                                                                           '-----------------'
    ),

    [_RAISE] = LAYOUT(
     //.--------------------------------------------.                     .--------------------------------------------.
        KC_1,    KC_2,    KC_3,    KC_4,    KC_5,                          KC_6,    KC_7,    KC_8,    KC_9,    KC_0,
     //|--------+--------+--------+--------+--------|                     |--------+--------+--------+--------+--------|
        KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,                         KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,
     //|--------+--------+--------+--------+--------|                     |--------+--------+--------+--------+--------|
        KC_F11,  KC_F12,  XXXXXXX, KANJI,   XXXXXXX,                       XXXXXXX, XXXXXXX, XXXXXXX, KC_UP,   KC_RO,
     //|--------+--------+--------+--------+--------+--------.   .--------+--------+--------+--------+--------+--------|
        KC_ESCT, KC_TBAL, KC_LGUI, LOWER,   KC_BSPC, KC_SPC,      KC_ENT,  KC_ENT,  KANJI,   KC_LEFT, KC_DOWN, KC_RGHT,
     //|--------+--------+--------+--------+--------+--------'   '--------+--------+--------+--------+--------+--------|
        KC_RSFT, LOWER,                                                                               RAISE,   KC_RCTL
     //'-----------------'                                                                           '-----------------'
    ),
```

File: keyboards/uranuma/keymaps/default/keymap.c [hunk 9]
```diff
@@ -0,0 +1,65 @@
+#include QMK_KEYBOARD_H
+
+enum layer_number {
+  _BASE,
+  _LOWER,
+  _RAISE,
+};
+
+enum custom_keycodes {
+  KANJI = SAFE_RANGE,
```
Comment (zvecr) [inline keyboards/uranuma/keymaps/default/keymap.c]: As a FYI, `KANJI` in this case doesnt actually do anything given there is no `process_record_user` for it.

**Comment by noroadsleft** [comment]
Thanks!

**Comment by yohewi** [comment]
Thanks‼︎

**Pull request merged**